### DISCUSSION TITLE
How do I add special Character constraint to MFXPasswordField

### DISCUSSION THREAD

**Discussion by noahheller** [palexdev/MaterialFX#218] [Archived]
I want the user to have to enter special characters in their password. I am able to a constraint of the number of characters with the following code:  `passwordField.getValidator().constraint("The password must be at least 8 characters long", passwordField.textProperty().length().greaterThanOrEqualTo(8));`. I can not figure out how to create my own `BooleanExpression `constraint.
-Thanks

**Comment by palexdev** [comment]
@goatfanboi23 There's a full example here [Demo](https://github.com/palexdev/MaterialFX/blob/main/demo/src/main/java/io/github/palexdev/materialfx/demo/controllers/TextFieldsController.java)
